examples/light-switch-app/nrfconnect/main/LightSwitch.cpp

```diff
@@ -80,3 +86,19 @@ void LightSwitch::DimmerChangeBrightness()
         DeviceLayer::PlatformMgr().ScheduleWork(BindingHandler::SwitchWorkerHandler, reinterpret_cast<intptr_t>(data));
     }
 }
+
+void LightSwitch::GenericSwitchInitialPress()
+{
+    uint8_t newPosition = 1;
+
+    Clusters::Switch::Attributes::CurrentPosition::Set(mLightGenericSwitchEndpointId, newPosition);
+    Clusters::SwitchServer::Instance().OnInitialPress(mLightGenericSwitchEndpointId, newPosition);
+}
+
+void LightSwitch::GenericSwitchReleasePress()
+{
+    uint8_t previousPosition = 0;
+
+    Clusters::Switch::Attributes::CurrentPosition::Set(mLightGenericSwitchEndpointId, previousPosition);
+    Clusters::SwitchServer::Instance().OnShortRelease(mLightGenericSwitchEndpointId, previousPosition);
```

Comment:
This is supposed to be handed the previous position, right?  As in, the value of `CurrentPosition` before we set it here.

Reply:
See above. This is now the releasedPosition which is 0.

Spec:

> CurrentPosition value 0 SHALL be assigned to the default position of the
> switch: for example the "open" state of a rocker switch, or the "idle" state of a push button switch.

The parameter however is called previousPosition in `void SwitchServer::OnShortRelease(EndpointId endpoint, uint8_t previousPosition)`, which should probably be changed then as well.

Reply:
No the point is that for a release event the field provided with the event in the spec is the position _before_ the release.  So the parameter name is in fact correct.

Reply:
Closing this conversation given @bzbarsky-apple approval. Created issue https://github.com/project-chip/connectedhomeip/issues/19923 for the linux part.